addon/mixins/ajax-request.js

```diff
@@ -225,15 +225,20 @@ export default Mixin.create({
    */
   request(url, options) {
     const hash = this.options(url, options);
-    return new Promise((resolve, reject) => {
-      this._makeRequest(hash)
-        .then(({ response }) => {
-          resolve(response);
-        })
-        .catch(({ response }) => {
-          reject(response);
-        });
+    const internalPromise = this._makeRequest(hash);
+
+    const ajaxPromise = new AJAXPromise((resolve, reject) => {
+      internalPromise.then(({ response }) => {
+        resolve(response);
+      })
+      .catch(({ response }) => {
+        reject(response);
+      });
     }, `ember-ajax: ${hash.type} ${hash.url} response`);
```

Comment:
@taras How important do you think this extra Promise label is?  This method could be simplified to just:

```js
return this.raw(...arguments)
  .then(({ response }) => response)
  .catch(({ response }) => {
    throw response;
  });
```

without it.  The internal request promise will still have the label; is adding `response` to the end of it really that useful?